templates/single_album_page.html

```diff
@@ -0,0 +1,30 @@
+{% extends "bootstrap/base.html" %}
+
+{% block content %}
+<h2>{{ query.album_title }}</h2>
+<h1> recorded_by: {{ subquery.artist_name }}
+<img src="{{ query.album_art }}">
+<div> TODO: subquery for this artist and genre info</div>
```

Comment:
maybe this? 

select album.title, album.album_cover, album.release_date, album.publisher, album.spotify_url, artist.name, genre.name as genre_name from album inner join artist on album.artist_id=artist.id inner join album_genre on album.id = album_genre.album_id inner join genre on album_genre.genre_id = genre.id where album.id = <ID>;

Reply:
this will work if reviews exist, but it won't if reviews don't work:
select   album.title,   album.album_cover,   album.release_date,   album.publisher,   album.spotify_url,   artist.name,   genre.name as genre_name,   review.rating from   album as album   inner join artist as artist on album.artist_id = artist.id   inner join album_genre on album.id = album_genre.album_id   inner join genre as genre on album_genre.genre_id = genre.id   inner join review on album.id = review.album_id where   album.id = 2;

Maybe do a touch first? something like
if review_table.check_for_any_reviews(album_id = 2):
    execute query to include ratings
else:
    execute query that just gets the other stuff

Reply:
I bet the first one won't work if there are no genre's included. When an album is created need to make sure that genre is a required field.

Reply:
Something to keep in mind with the review query where there are multiple genres associated: 
the resulting table will yield one row per genre per review ratings so any summing needs to keep that in mind. for example, where there is only one review on an album where id = 3:

`select   album.title,   album.album_cover,   album.release_date,   album.publisher,   album.spotify_url,   artist.name,   genre.name as genre_name,   review.rating from   album as album   inner join artist as artist on album.artist_id = artist.id   inner join album_genre on album.id = album_genre.album_id   inner join genre as genre on album_genre.genre_id = genre.id   inner join review on album.id = review.album_id where   album.id = 3;
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
| title          | album_cover                                  | release_date | publisher | spotify_url                                           | name             | genre_name | rating |
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Folk       |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Acoustic   |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Punk       |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Anarchist  |     10 |
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
`

but when i put in two reviews:

`select   album.title,   album.album_cover,   album.release_date,   album.publisher,   album.spotify_url,   artist.name,   genre.name as genre_name,   review.rating from   album as album   inner join artist as artist on album.artist_id = artist.id   inner join album_genre on album.id = album_genre.album_id   inner join genre as genre on album_genre.genre_id = genre.id   inner join review on album.id = review.album_id where   album.id = 3;
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
| title          | album_cover                                  | release_date | publisher | spotify_url                                           | name             | genre_name | rating |
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Folk       |      4 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Folk       |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Acoustic   |      4 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Acoustic   |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Punk       |      4 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Punk       |     10 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Anarchist  |      4 |
| Live the dream | https://f4.bcbits.com/img/a3617947760_10.jpg | 2015-05-12   | Plan-It-X | https://open.spotify.com/album/6LoWh2KXtS4sVhFLgeT7Yo | Ramshackle Glory | Anarchist  |     10 |
+----------------+----------------------------------------------+--------------+-----------+-------------------------------------------------------+------------------+------------+--------+
`